### DISCUSSION TITLE
When `output` is `export`, the build fails

### DISCUSSION THREAD

**Discussion by kjxbyz** [amannn/next-intl#628] [Q&A]
### Description

```
StaticGenBailoutError: Page with `dynamic = "error"` couldn't be rendered statically because it used `headers`. See more info here: https://nextjs.org/docs/app/building-your-application/rendering/static-and-dynamic#dynamic-rendering
--
00:51:10.529 | at staticGenerationBailout (/vercel/path0/app/.next/server/chunks/570.js:1:81342)
00:51:10.529 | at headers (/vercel/path0/app/.next/server/chunks/41.js:13:49131)
00:51:10.529 | at /vercel/path0/app/.next/server/chunks/41.js:13:44243
00:51:10.529 | at /vercel/path0/node_modules/.pnpm/next@14.0.1_@babel+core@7.23.2_@opentelemetry+api@1.6.0_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:108:5119
00:51:10.529 | at s (/vercel/path0/app/.next/server/chunks/41.js:13:44967)
00:51:10.529 | at e (/vercel/path0/app/.next/server/chunks/41.js:13:45059)
00:51:10.529 | at m (/vercel/path0/app/.next/server/chunks/41.js:13:46804)
00:51:10.529 | at em (/vercel/path0/node_modules/.pnpm/next@14.0.1_@babel+core@7.23.2_@opentelemetry+api@1.6.0_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:76:10452)
00:51:10.529 | at /vercel/path0/node_modules/.pnpm/next@14.0.1_@babel+core@7.23.2_@opentelemetry+api@1.6.0_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:76:22148
00:51:10.530 | at Object.toJSON (/vercel/path0/node_modules/.pnpm/next@14.0.1_@babel+core@7.23.2_@opentelemetry+api@1.6.0_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:76:25726) {
00:51:10.530 | code: 'NEXT_STATIC_GEN_BAILOUT'
```

### Mandatory reproduction URL (CodeSandbox or GitHub repository)

https://github.com/cyf/faforever-next/pull/72

### Reproduction description

Steps to reproduce:
1. git clone https://github.com/cyf/faforever-next
2. git fetch
3. git checkout origin/renovate/next-intl-3.x -b renovate/next-intl-3.x
4. pnpm i
5. pnpm build:app

### Expected behaviour

When `output` is `export`, the build is normal

**Comment by amannn** [comment]
Please see [Usage without middleware (static export)](https://next-intl-docs.vercel.app/docs/routing/middleware#usage-without-middleware-static-export) in the docs, this should help. I'll move this to a discussion since it seems to be a usage question.

**Reply by kjxbyz** [reply]
This error will only be reported after upgrading to a new version. Version 2.22.1 works fine

**Reply by amannn** [reply]
Oh, so you've previously used `next-intl` only in Client Components? That's interesting, I'll see if I can have a look later.

**Reply by amannn** [reply]
I can confirm there's something wrong. I've reopened this as https://github.com/amannn/next-intl/issues/631, thanks for the report!

**Reply by amannn** [reply]
@kjxbyz This has been addressed in https://github.com/amannn/next-intl/pull/633 (`next-intl@3.0.3`). There is however a change necessary that was missing in the release notes previously, please see [Changes to `NextIntlClientProvider`](https://next-intl-docs.vercel.app/blog/next-intl-3-0#changes-to-nextintlclientprovider).

Hope this helps!

**Reply by flowerofmycountry** [reply]
Can you provide an example with 'output: "export"' ?
